subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4697,6 +4697,16 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			lwm2m_engine_context_close(client_ctx);
 			return -errno;
 		}
+
+#if defined(CONFIG_MBEDTLS_SERVER_NAME_INDICATION)
```

Comment:
I'm not sure, is this check needed? It should be no harm to set hostname in either case. Perhaps just verify that `client_ctx->desthostname != NULL` (though it seems there's no way socket is open before hostname is actually parsed.